composer.json

```
@@ -5,16 +5,16 @@
     "license": "MIT",
     "require": {
         "php": "~8.1",
-        "jangregor/phpstan-prophecy": "^1.0",
-        "phpmd/phpmd": "^2.13",
-        "phpstan/phpdoc-parser": "^1.12",
-        "phpstan/phpstan": "^1.8.11",
-        "rector/rector": "^0.14.6",
-        "slevomat/coding-standard": "^8.6.2",
-        "symplify/easy-coding-standard": "^11.1.10",
-        "symplify/phpstan-extensions": "^11.1.10",
-        "symplify/phpstan-rules": "^11.1.10",
-        "nette/utils": "^3.2.8"
+        "jangregor/phpstan-prophecy": "1.0.0",
+        "phpmd/phpmd": "2.13.0",
+        "phpstan/phpdoc-parser": "1.13.0",
```

```suggestion
        "phpstan/phpdoc-parser": "^1.13.0",
```

Otherwise there is no way to install more recent versions of dependencies without patching this file.

---

Hi @andrew-demb,
While working on this package for a few years, we've found a lot of cases when dependencies of the package conflict with the dependencies of a project EasyQuality installed in. Especially the Symfony packages: you can use 5.x in a project but EasyQuality installs 6.x.

So, the idea is to always install EasyQuality separately, not within the main `composer.json` file of a project. Simply create a new folder in your project and install EasyQuality into it. This way you will never experience dependency conflicts.

---

Hi @roman-eonx.
I'm not saying about dependency conflicts, but rather about allowing users to install newer dependency versions that are semver-compatible with your package.

If tomorrow there will be released `phpstan/phpdoc-parser:1.13.1` fix critical bugfix for some code that is not common for your codebase (but can be commonly used by users of `eonx-com/easy-quality`) - there is no way for users (that installed `eonx-com/easy-quality` in a separated directory as recommended in readme) to get this bugfix from `phpstan/phpdoc-parser:1.3.1` before `eonx-com/easy-quality` will be released with updated version restriction.

---

I prefer to consider this package as final and not allow dependency updates.

I had faced situations when Rector was silently not working, and we had legacy code in our projects for a few months, and we had many comments in reviews like "Please write as ...", "Please use this instead this", etc. Not a big deal, but just spending time on things we think the Rector must fix.

One more point - we want to be sure that Quality Tools work the same way in all our projects. That's why we introduce our config sets under `/config`. This also allows us to distribute the most interesting features across all projects with minimal work, don't care to match the dependencies version.

In the past, we have fixed versions of packages. Later, I made flexible versions like a "package/name": "^1.0.0", and now I realise it is better to have fixed versions.

@andrew-demb, we are delighted that you are using our package, and we are working on its improvements.

If you need an extra dependency upgrade, please make PR or open an issue, and we will try to manage this asap.

If you are faced with other bugs concerning dependencies and their versions - I will be glad to discuss this and find the right solution.

And the last point is that I am very interested in seeing our package use cases in other projects and sharing our experience. I believe this will help to provide better products.